pkgs/by-name/pr/proton-ge-custom/package.nix

```diff
@@ -0,0 +1,53 @@
+{ lib
+, stdenvNoCC
+, fetchurl
+, writeScript
+}:
+stdenvNoCC.mkDerivation (finalAttrs: {
+  pname = "proton-ge-bin";
+  version = "GE-Proton9-1";
+
+  src = fetchurl {
+    url = "https://github.com/GloriousEggroll/proton-ge-custom/releases/download/${finalAttrs.version}/${finalAttrs.version}.tar.gz";
+    hash = "sha256-wCIffeayOy3kEwmIKB7e+NrliuSpKXoVYC334fxVB3U=";
+  };
+
+  buildCommand = ''
+    runHook preBuild
+    mkdir -p $out/{bin,opt}
+    tar -C $out/opt --strip=1 -x -f $src
+    ln -s $out/opt/toolmanifest.vdf $out/bin/toolmanifest.vdf
+    install -Dm644 $out/opt/compatibilitytool.vdf $out/bin/compatibilitytool.vdf
```

Comment:
Alright, so what this does is create a structure like this:

```
result/
├── bin
│   ├── compatibilitytool.vdf
│   └── toolmanifest.vdf -> /nix/store/397njgf44bqpqaf7dk9q272z17nhmbjb-proton-ge-custom/opt/toolmanifest.vdf
└── opt
    ├── compatibilitytool.vdf
    ├── filelock.py
    ├── files
    ├── LICENSE
    ├── LICENSE.OFL
    ├── PATENTS.AV1
    ├── proton
    ├── proton_3.7_tracked_files
    ├── protonfixes
    ├── toolmanifest.vdf
    ├── user_settings.sample.py
    └── version
```

I don't understand what this is intended to do.

Neither of the files in bin are executables and rather than a subdirectory, the contents are placed in `opt/` directly?

I would have expected the output of this to look like this instead:

```
result/
├── compatibilitytool.vdf
├── toolmanifest.vdf -> /nix/store/397njgf44bqpqaf7dk9q272z17nhmbjb-proton-ge-custom/toolmanifest.vdf
└── GE-Proton9-1
    ├── compatibilitytool.vdf
    ├── filelock.py
    ├── files
    ├── LICENSE
    ├── LICENSE.OFL
    ├── PATENTS.AV1
    ├── proton
    ├── proton_3.7_tracked_files
    ├── protonfixes
    ├── toolmanifest.vdf
    ├── user_settings.sample.py
    └── version
```

but I don't know how steam interprets this. From the "documentation" I'd assume that the search path includes directories to be searched; just like `~/.steam/root/compatibilitytools.d/` is searched.

Another topic I'd like to outline is whether all of this should rather be in a special output or have a `usr/share/steam/compatibilitytools.d/` inside the package. I'd gravitate towards having a separate `compatibilitytool` output where the files are arranged like the second tree which the option would select.

Reply:
I don't think a special output (or a usr/share...) path is necessary. Steam cat pick up the compat tool from the manifest directly.

Reply:
Parts of the structure done here is a result of https://github.com/NixOS/nixpkgs/pull/189398#discussion_r961805387 and  https://github.com/NixOS/nixpkgs/pull/189398#issuecomment-1612019240 and the follow suggustion to use [makeBinPath](https://nixos.org/manual/nixpkgs/stable/#function-library-lib.strings.makeBinPath) for path creation.
I initially suggested on the extraCompatPackages pr to not assume that there is a /bin folder.
That would it make possible to do your expected result (and would be more similar what i had initially when i did first tests with that before  https://github.com/NixOS/nixpkgs/pull/189398 was raised.

When not using makeBinPath we either can drop the whole package content at $out, or drop it at $out/<something> like suggested. It would be just needed then that compatibilitytool.vdf  and  toolmanifest.vdf are set correctly.

The overall requirement from steam is that every path listed in extraCompatPackages has to have compatibilitytool.vdf sitting there and its install_path to point at a toolmanifest.vdf (AFAIK).
So as we are now adding the first compat tool here on my opinion we just to have align the steam module to comply with what we want to do here. 

I am personally up here for any change, dropping everything just in $out might be the easiest but was kind of declined on the other PR by https://github.com/NixOS/nixpkgs/pull/189398#discussion_r962180579

Reply:
I am still in favor of keeping the current structure alongside fetchurl to avoid linking/copying manually.